app/src/main/java/com/github/se/gomeet/viewmodel/EventViewModel.kt

```diff
@@ -246,6 +246,8 @@ class EventViewModel(private val creatorId: String? = null) : ViewModel() {
     Log.d("CreatorID", "Creator ID is $creatorId")
     CoroutineScope(Dispatchers.IO).launch {
       try {
+        val participantsWithCreator =
```

Comment:
I don't get how the `participants` field of an event could not have the `creator` of the event inside the list. I think we implemented the logic such that the `participants` always contain the `creator` of the event. Am I wrong ? If so, what you are doing is good

Reply:
It's on a just-in-case basis. Logically it should always contain the creator of the event, but if createEvent is called  with the wrong list of participants somewhere then it's a fail-safe way of making sure that the creator is in the participants list while making sure that the creator doesn't get added a second time. Thinking of it now, maybe the participants lists (all of them) should be a set not a list, since they shouldn't contain duplicates.

Reply:
Yes that's right, that is definitely something we can do later for M4, good catch !